### PR TITLE
[ESLint rule]: recoil value and setter should be named after their at…

### LINKED ISSUE #1372: [ESLint rule]: recoil value and setter should be named after their atom name
NOT OK : 

```
const [myVariable, setMyVariable] = useRecoilState(somethingElseState)
```

OK : 

```
const [myVariable, setMyVariable] = useRecoilState(myVariableState)
```

It should work with useRecoilState, useRecoilValue, useRecoilScopedState, etc.
```[tasklist]
### Tasks
- [ ] Create a custom eslint-plugin-twenty rule similar to `styled-components-prefixed-with-styled`
```

### PR DESCRIPTION
**Pull request by gitstart-twenty**
## USAGE

1. Reinstall the plugin
    ```
    cd front
    yarn add file:../packages/eslint-plugin-twenty --dev
    ```
2. Restart the eslint extension server for vscode
    Use the VSCode command palette (`Cmd + Shift + P` or `Ctrl + Shift + P`) and run `ESLint: Restart ESLint Server`.

Fixes #1372

### CODE REVIEW

File: packages/eslint-plugin-twenty/rules/matching-state-variable.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+          node.init.callee.name.startsWith("useRecoil")
+        ) {
+          const stateNameBase =
+            node.init.arguments[0] && node.init.arguments[0].name;
+
+          if (!stateNameBase) {
+            return;
+          }
+
+          const expectedVariableNameBase = stateNameBase.replace("State", "");
```
Comment (lucasbordeau) [inline packages/eslint-plugin-twenty/rules/matching-state-variable.js:31]: Could you please try to also cover : 

- State
- FamilyState
- Selector
- FamilySelector
- ScopedState

If it's too much work we can do that in another PR

File: packages/eslint-plugin-twenty/rules/matching-state-variable.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    schema: [],
+  },
+  create: function (context) {
+    return {
+      VariableDeclarator(node) {
+        if (
+          node.init &&
+          node.init.callee &&
+          node.init.callee.name &&
+          node.init.callee.name.startsWith("useRecoil")
```
Comment (Weiko) [inline packages/eslint-plugin-twenty/rules/matching-state-variable.js:22]: We can probably use optional chaining here?

**Review by charlesBochet — APPROVED** [review]

**Comment by charlesBochet** [comment]
Thank you!
